File: go/mbpay/order_info.go
package mbpay

import (
	"fmt"
	"net/http"
	"strconv"
)

// GetOrderInfo 查询订单信息
func (c *Client) GetOrderInfo(orderNo string, merchantID int64) (*OrderInfoResponse, error) {
	// 参数验证
	if orderNo == "" {
		return nil, fmt.Errorf("order_no is required")
	}
	if merchantID <= 0 {
		return nil, fmt.Errorf("merchant_id is required and must be greater than 0")
	}

	// 构建请求参数
	params := map[string]string{
		"order_no":    orderNo,
		"merchant_id": strconv.FormatInt(merchantID, 10),
	}

	// 执行请求
	resp, err := c.doRequest(http.MethodPost, "/merchant/orderinfo", params)
	if err != nil {
		return nil, err
	}

	// 检查业务错误
	if !resp.IsSuccess() {
		return nil, resp.ToError()
	}

	// 解析 data 字段
	orderResp := &OrderInfoResponse{}

	if orderNo, ok := resp.Data["order_no"].(string); ok {
		orderResp.OrderNo = orderNo
	} else {
		return nil, fmt.Errorf("invalid order_no format in response")
	}

	if platformOrderNo, ok := resp.Data["platform_order_no"].(string); ok {
		orderResp.PlatformOrderNo = platformOrderNo
	} else {
		orderResp.PlatformOrderNo = "" // 可能为空
	}

	if amount, ok := resp.Data["amount"].(float64); ok {
		orderResp.Amount = int64(amount)
	} else if amount, ok := resp.Data["amount"].(int64); ok {
		orderResp.Amount = amount
	} else if amount, ok := resp.Data["amount"].(int); ok {
		orderResp.Amount = int64(amount)
	} else {
		return nil, fmt.Errorf("invalid amount format in response")
	}

	if platformFee, ok := resp.Data["platform_fee"].(float64); ok {
		orderResp.PlatformFee = int64(platformFee)
	} else if platformFee, ok := resp.Data["platform_fee"].(int64); ok {
		orderResp.PlatformFee = platformFee
	} else if platformFee, ok := resp.Data["platform_fee"].(int); ok {
		orderResp.PlatformFee = int64(platformFee)
	} else {
		orderResp.PlatformFee = 0 // 默认为 0
	}

	if status, ok := resp.Data["status"].(float64); ok {
		orderResp.Status = int(status)
	} else if status, ok := resp.Data["status"].(int); ok {
		orderResp.Status = status
	} else {
		return nil, fmt.Errorf("invalid status format in response")
	}

	if statusText, ok := resp.Data["status_text"].(string); ok {
		orderResp.StatusText = statusText
	} else {
		orderResp.StatusText = "" // 可能为空
	}

	if expiresAt, ok := resp.Data["expires_at"].(string); ok {
		orderResp.ExpiresAt = expiresAt
	} else {
		orderResp.ExpiresAt = "" // 可能为空
	}

	if createdAt, ok := resp.Data["created_at"].(string); ok {
		orderResp.CreatedAt = createdAt
	} else {
		orderResp.CreatedAt = "" // 可能为空
	}

	if paidAt, ok := resp.Data["paid_at"].(string); ok {
		orderResp.PaidAt = paidAt
	} else {
		orderResp.PaidAt = "" // 可能为空
	}

	return orderResp, nil
}

File: python/mbpay/order_info.py
"""订单信息查询功能"""

from .client import Client
from .types import OrderInfoResponse, Response, MBPayError


def get_order_info(self: Client, order_no: str, merchant_id: int) -> OrderInfoResponse:
    """
    查询订单信息
    
    Args:
        order_no: 商户订单号（必填）
        merchant_id: 商户ID（必填）
        
    Returns:
        OrderInfoResponse: 订单信息响应
        
    Raises:
        MBPayError: API 业务错误
        requests.RequestException: 网络请求错误
    """
    # 参数验证
    if not order_no:
        raise MBPayError(0, "order_no is required")
    if merchant_id <= 0:
        raise MBPayError(0, "merchant_id is required and must be greater than 0")
    
    # 构建请求参数
    params = {
        "order_no": order_no,
        "merchant_id": str(merchant_id),
    }
    
    # 执行请求
    resp: Response = self._do_request("POST", "/merchant/orderinfo", params)
    
    # 解析 data 字段
    data = resp.data
    order_no_resp = data.get("order_no", "")
    platform_order_no = data.get("platform_order_no", "")
    amount = int(data.get("amount", 0))
    platform_fee = int(data.get("platform_fee", 0))
    status = int(data.get("status", 0))
    status_text = data.get("status_text", "")
    expires_at = data.get("expires_at", "")
    created_at = data.get("created_at", "")
    paid_at = data.get("paid_at", "")
    
    if not order_no_resp:
        raise MBPayError(0, "invalid order_no format in response")
    
    return OrderInfoResponse(
        order_no=order_no_resp,
        platform_order_no=platform_order_no,
        amount=amount,
        platform_fee=platform_fee,
        status=status,
        status_text=status_text,
        expires_at=expires_at,
        created_at=created_at,
        paid_at=paid_at
    )


# 将方法绑定到 Client 类
Client.get_order_info = get_order_info










